src/xref2/errors.ml

```diff
@@ -129,6 +126,35 @@ module Tools_error = struct
     | `Fragment_root -> Format.fprintf fmt "Fragment root"
 end
 
+ (* Ugh. we need to determine whether this was down to an unexpanded module type error. This is horrendous. *)
+ let is_unexpanded_module_type_of =
```

Comment:
This only need to work on `signature_of_module_error` and to recurse into `parent_lookup_error` and `` [> `Parent of parent_lookup_error ] `` (the only interesting part of other `*_error`). For example:
```ocaml
let is_unexpanded_module_type_of =
  let rec signature_of_module = function
    | `UnexpandedTypeOf _ -> true
    | `OpaqueModule | `UnresolvedForwardPath -> false
    | `UnresolvedPath (`Module (_, e)) -> rec_parent e
    | `UnresolvedPath (`ModuleType (_, e)) -> rec_parent e
  and rec_parent = function
    | `Parent p -> parent p
    | _ -> false
  and parent = function
    | `Parent_sig e -> signature_of_module e
    | `Parent_module_type p -> rec_parent p
    | `Parent_expr p -> rec_parent p
    | `Parent_module p -> rec_parent p
    | `Fragment_root -> false
    | `Parent p -> parent p
  in
  signature_of_module
```

Also, I don't think `parent_lookup_error` should be part of `any`.

Reply:
aren't they all errors?

Reply:
`parent_lookup_error` are not errors I think, they're just an indirection on top of `` `Parent `` (present in every kinds of error).